velox/type/tz/TimeZoneMap.cpp

```diff
@@ -44,7 +49,23 @@ TTimeZoneDatabase buildTimeZoneDatabase(
   tzDatabase.resize(dbInput.back().first + 1);
 
   for (const auto& entry : dbInput) {
-    tzDatabase[entry.first] = std::make_unique<std::string>(entry.second);
+    std::unique_ptr<TimeZone> timeZonePtr;
+
+    if (entry.first == 0) {
+      timeZonePtr = std::make_unique<TimeZone>(
+          "UTC", entry.first, date::locate_zone("UTC"));
+    } else if (entry.first <= 1680) {
+      std::chrono::minutes offset = getTimeZoneOffset(entry.first);
+      timeZonePtr =
+          std::make_unique<TimeZone>(entry.second, entry.first, offset);
+    }
+    // Every single other time zone entry (outside of offsets) needs to be
+    // available in external/date or this will throw.
+    else {
+      timeZonePtr = std::make_unique<TimeZone>(
+          entry.second, entry.first, date::locate_zone(entry.second));
+    }
```

Comment:
It seems that `date::locate_zone` requires the time-zone to be recognizable in OS? 

I am not sure but probably this change causes Gluten's CentOS 8 tests to throw

```
Europe/Kyiv not found in timezone database
```

[link](https://productionresultssa6.blob.core.windows.net/actions-results/1b339a8c-794d-4a8a-96f6-3c470b24e622/workflow-job-run-072e938f-deca-5370-c3dd-1d809009c6e1/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-07-29T04%3A08%3A56Z&sig=diDa%2FEy0bAllOgvpoKEb9I2UUnRub8LiDTPMbqzdWAI%3D&ske=2024-07-29T15%3A51%3A48Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-07-29T03%3A51%3A48Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2024-05-04&sp=r&spr=https&sr=b&st=2024-07-29T03%3A58%3A51Z&sv=2024-05-04)

Can we loose the restriction here?

Reply:
@pedroerp

Reply:
@zhztheplayer I guess you are using an outdated version of the tzdata database in your system. Could you update it? Which version are you using? 

You can check with something like

```
$ rpm -qa | grep tzdata
tzdata-2024a-2.el9.noarch
```

Reply:
It's by default `tzdata-2021a-1.el8.noarch` in official `centos:8` docker image. So I need to update it probably.

```bash
[root@47881716f3de /]# rpm -qa | grep tzdata
tzdata-2021a-1.el8.noarch
[root@47881716f3de /]# ls /usr/share/zoneinfo/Europe/
Amsterdam  Belfast     Brussels   Chisinau    Guernsey     Jersey       Lisbon      Madrid     Monaco   Paris      Rome        Saratov     Stockholm  Ulyanovsk  Vienna     Zagreb
Andorra    Belgrade    Bucharest  Copenhagen  Helsinki     Kaliningrad  Ljubljana   Malta      Moscow   Podgorica  Samara      Simferopol  Tallinn    Uzhgorod   Vilnius    Zaporozhye
Astrakhan  Berlin      Budapest   Dublin      Isle_of_Man  Kiev         London      Mariehamn  Nicosia  Prague     San_Marino  Skopje      Tirane     Vaduz      Volgograd  Zurich
Athens     Bratislava  Busingen   Gibraltar   Istanbul     Kirov        Luxembourg  Minsk      Oslo     Riga       Sarajevo    Sofia       Tiraspol   Vatican    Warsaw
```

Reply:
yeah, I suppose a simple update to the tzdata package should solve it. We could also do what you suggested above; allow timezone without a tzdata database, and throw at runtime if they try to be used. If you think that's useful, could you open an Issue?

Reply:
https://github.com/facebookincubator/velox/pull/10654